Looks good, introduces one new test failure though:
```
    @pytest.mark.parametrize(
        "value",
        (
            1,
            1.3,
            "foo",
            True,
            b"binary",
        ),
    )
    def test_table_default_values(fresh_db, value):
        fresh_db["default_values"].insert(
            {"nodefault": 1, "value": value}, defaults={"value": value}
        )
        default_values = fresh_db["default_values"].default_values
>       assert default_values == {"value": value}
E       AssertionError: assert {'value': 'True'} == {'value': True}
E         
E         Differing items:
E         {'value': 'True'} != {'value': True}
E         Use -v to get more diff
```

--- a/docs/changelog.rst
+++ b/docs/changelog.rst
@@ -42,6 +42,10 @@ Column names passed to Python API methods are now matched against the table sche
 - Foreign key columns are validated and recorded using the casing of the actual schema columns, in ``foreign_keys=`` when creating tables, ``db.add_foreign_keys()``, ``table.add_foreign_key()`` and ``table.add_column(fk_col=...)``. Duplicate foreign key detection is also case-insensitive.
 - ``table.create()`` with ``pk=``, ``not_null=``, ``defaults=`` or ``column_order=`` referencing columns using different casing no longer creates an unwanted extra primary key column or raises a ``ValueError``.
 
+Everything else:
+
+- Fixed a bug where ``table.transform()`` could convert ``DEFAULT TRUE``, ``DEFAULT FALSE`` and ``DEFAULT NULL`` column defaults into quoted string defaults when rebuilding a table. Thanks, `Vincent Gao <https://github.com/gaoflow>`__. (`#764 <https://github.com/simonw/sqlite-utils/pull/764>`__)
+
 .. _v4_0rc2:
 
 4.0rc2 (2026-07-04)

--- a/sqlite_utils/db.py
+++ b/sqlite_utils/db.py
@@ -710,7 +710,7 @@ class Database:
         if str(value).upper() in ("CURRENT_TIME", "CURRENT_DATE", "CURRENT_TIMESTAMP"):
             return value
 
-        if str(value).upper() in ("TRUE", "FALSE", "NULL"):
+        if isinstance(value, str) and value.upper() in ("TRUE", "FALSE", "NULL"):
             # Keyword literals must stay unquoted; quoting them would turn the
             # default into a string ('TRUE' instead of 1, 'NULL' instead of null).
             return value

